### ISSUE TITLE
inconsistent behavior when re-assigning variable

### ISSUE THREAD

**Issue by cshesse** [microsoft/pyright#363]
**Describe the bug**

Re-assigning a variable has inconsistent behavior when the variable is a function.

**To Reproduce**

```
# pyright: strict

# 1) fails
# def i():
#     pass
# i = 1

# 2) OK
# def i():
#     pass
# i: int = 1

# 3) fails, but with confusing error message
# i: int = 1
# i: float = 1.1
```

1)  I expect this to fail and it does:

```
pyright --project .
Loading configuration file at C:\p\misc\pyright_test\pyright_test14\pyrightconfig.json
typingsPath C:\p\misc\pyright_test\pyright_test14\typings is not a valid directory.
Searching for source files
Found 1 source file
C:\p\misc\pyright_test\pyright_test14\test.py
  8:5 - error: Expression of type 'Literal[1]' cannot be assigned to declared type '() -> None'
  'Literal[1]' cannot be assigned to '() -> None'
1 error, 0 warnings
Completed in 1.17sec
```

2) I expect this to fail in a similar way to 1, but it does not, no error is produced:

```
pyright --project .
Loading configuration file at C:\p\misc\pyright_test\pyright_test14\pyrightconfig.json
typingsPath C:\p\misc\pyright_test\pyright_test14\typings is not a valid directory.
Searching for source files
Found 1 source file
0 errors, 0 warnings
Completed in 1.177sec
```

3) I expect this to fail, and it does:

```
pyright --project .
Loading configuration file at C:\p\misc\pyright_test\pyright_test14\pyrightconfig.json
typingsPath C:\p\misc\pyright_test\pyright_test14\typings is not a valid directory.
Searching for source files
Found 1 source file
C:\p\misc\pyright_test\pyright_test14\test.py
  14:4 - error: Declared type 'int' is not compatible with declared type 'float'
1 error, 0 warnings
Completed in 1.175sec
```

**VS Code extension or command-line**

```
pyright --version
pyright 1.1.5
```

**Comment by erictraut** [comment]
Thanks for the suggestion. I've improved the error messages for type declaration collisions.

**Comment by erictraut** [comment]
This is fixed in 1.1.6, which I just published.
